packages/auth/src/authClient.ts

```diff
@@ -380,34 +382,97 @@ export class AuthClient {
 
   /**
    * Initiates passkey (WebAuthn) authentication flow.
+   *
    * This method uses WebAuthn standards to authenticate users with biometric or hardware security keys.
+   * Can be used for both new passkey creation (when credentialId is undefined) and
+   * authentication with existing passkeys (when credentialId is provided).
    *
+   * @param {string} [credentialId] - Optional credential ID for authenticating with existing passkey
    * @returns {Promise<AuthSession>} Promise that resolves to an auth session instance
-   * @throws {Error} Currently throws "Not implemented" as this method is under development
    *
    * @example
-   * ```ts
+   * ```ts twoslash
+   * // New passkey authentication
    * const authSession = await authClient.loginWithPasskey();
+   *
+   * // Authenticate with existing passkey
+   * const authSession = await authClient.loginWithPasskey("existing-credential-id");
    * ```
    */
-  public async loginWithPasskey(): Promise<AuthSession> {
+  public async loginWithPasskey(credentialId?: string): Promise<AuthSession> {
     // TODO: figure out what the current passkey code is doing.
+    // For new passkey authentication, credentialId would be undefined initially
     const stamper = await this.createWebAuthnStamper({
-      credentialId: undefined,
+      credentialId,
     });
     return notImplemented(stamper);
   }
 
+  /**
+   * Creates an instance of AuthSession from a previously saved authentication session state that has not expired.
+   *
+   * This method takes a JSON string representation of a serialized AuthSessionState (typically obtained
+   * from AuthSession.getAuthSessionState()) and attempts to restore the authentication session.
+   * The method will validate the session expiration and handle different authentication types appropriately.
+   *
+   * @param {string} state - The serialized authentication session state as a JSON string
+   * @returns {Promise<AuthSession | undefined>} A promise that resolves to an AuthSession instance if the session is valid and not expired, undefined if expired
+   *
+   * @throws {Error} Throws an error if the session state is invalid or if credential injection fails
+   * @throws {Error} For passkey authentication, throws if credentialId is missing from the session state
+   *
+   * @example
+   * ```ts twoslash
+   * // Restore a session from stored JSON string
+   * const sessionJson = localStorage.getItem('authSession');
+   * if (sessionJson) {
+   *   const authSession = await authClient.loadAuthSessionState(sessionJson);
+   *   if (authSession) {
+   *     console.log('Session restored successfully');
+   *   } else {
+   *     console.log('Session expired');
+   *   }
+   * }
+   * ```
+   */
+  public async loadAuthSessionState(
```

Comment:
can we rename to `restoreAuthSession()? It's a clearer intent that you're deserializing/restoring a session. the state is something that we pass in in order to restore, but not something that we're loading